src/Maestro/Maestro.Web/Helpers.cs

```diff
@@ -99,5 +101,60 @@ public static async Task<IActionResult> ProxyRequestAsync(this HttpContext conte
             }
 
         }
+
+        public static Dictionary<string, KustoQuery> CreateBuildTimesQueries(string repository, string branch, int days)
+        {
+            var parameters = new List<KustoParameter> {
+                new KustoParameter("_Repository", KustoDataTypes.String,  repository.Split("/").Last()),
+                new KustoParameter("_SourceBranch", KustoDataTypes.String, branch),
+                new KustoParameter("_Days", KustoDataTypes.TimeSpan, $"{days}d")
+            };
+
+            string publicProject = "public";
+
+            // Builds in AzDo are only found in the internal project
+            if (repository.Contains("dev.azure.com"))
+            {
+                publicProject = "internal";
+            }
+
+            // We only care about builds that complete successfully or partially successfully 
+            // from the given repository. We summarize duration of the builds over the last 7 days.
+            // There are multiple different definitions that run in parallel, so we summarize
+            // on the definition id and ultimately choose the definition that took the longest.
+            string commonQueryText = @"| where Result != 'failed' and Result != 'canceled' 
+                | where FinishTime > ago(_Days) 
+                | extend duration = FinishTime - StartTime 
+                | summarize average_duration = avg(duration) by DefinitionId
+                | summarize max(average_duration)";
+
+            // We only want the pull request time from the public ci. We exclude on target branch,
```

Comment:
>public [](start = 59, length = 6)

I can't remember, did we decide to only pull PR times from the public project, even if the PR was an internal PR (e.g. security servicing build?)

Reply:
Dependency flow only goes through a public PR right (for github repos). Assuming yes, that is why I am using the public PR time, ignoring the internal PR time.

Reply:
It ends up going through internal PRs for security servicing releases. You actually might be already good here, because when dependency flow happens in internal builds, the repository will be from dev.azure.com, which means `publicProject` will be internal. Would need to try it out to be sure though.

Nit: might want to rename the variable since it can be both "public" or internal"

Reply:
renamed it.